shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -32,13 +36,16 @@ function shouldBlockHeaders (request, tab, requestIsTracker) {
  * @returns {{responseHeaders: Array<{name: string, value:string}>} | undefined}
  */
 function dropTracking3pCookiesFromResponse (request) {
+    if (request.type === 'main_frame' || request.responseHeaders.findIndex(header => header.name.toLowerCase() === 'set-cookie') === -1) {
```

Comment:
Sounds good to me to return here if it's not a main_frame request.

I'm not sure either way about checking for the set-cookie header here though. Maybe that would be worse than first checking the tab is known and the request is considered a tracker. Equally maybe it's better, up to you anyway.

Reply:
Looking for the set-cookie header is an array scan of max. 5-10 values. It also will likely excludes more than the other two checks, so doing it first means we have fewer tab and tracker lookups to do in total. My intuition is that this is the most performant check order.